up-l3/usubscription/v3/README.adoc

```diff
@@ -18,139 +18,410 @@ SPDX-FileType: DOCUMENTATION
 
-Subscribers are automatically registered to receive subscription change notifications for topics they have subscribed too through the Subscribe() API.
+[.specitem,oft-sid="dsn~usubscribe-change-notification~1",oft-needs="impl,utest"]
+--
+To support this use case, uSubscription service *SHALL* make all subscription changes available as event messages, published on a topic specifically defined for this purpose.
+--
 
-* Subscribers *MUST NOT* call the `RegisterForNotifications(`) as they are not permitted to receive notifications for subscription changes by other subscribers
-* Subscription change notification *MUST* be sent whenever the `SubscriptionState` changes for said subscriber (ex, `SUBSCRIBE_PENDING` → `SUBSCRIBED`)
+[.specitem,oft-sid="req~usubscribe-change-notification-type~1",oft-needs="impl,utest"]
+--
+Subscription change messages *MUST* be of type https://github.com/eclipse-uprotocol/up-spec/blob/da5ca97d3a7541d2fcd52ed010bc3bcca92e46cb/up-core-api/uprotocol/core/usubscription/v3/usubscription.proto#L302[Update].
+--
 
-=== Publishers
+[#usubscription-default-notification-topic]
+[.specitem,oft-sid="req~usubscribe-change-notification-topic~1",oft-needs="impl,utest"]
+--
+uSubscription service *SHALL* publish change notification messages to the topic
+ `/core.usubscription/3/subscriptions#Update`.
+--
 
-* *MUST* *NOT* be permitted to call `RegisterForNotifications()` for topics they do not produce, i.e. the uEntity sending the request is not equal to the `UEntity` of the request message's `UUri` parameter.
-* Subscription change notifications *MUST* be sent for changes to `SubscriptionState` for any subscriber that is subscribed to the topic
+=== Custom notification topics
 
-=== Dispatchers
+In addition to publishing subscription change notifications on the link:#usubscription-default-notification-topic[default notification channel], uEntities may want to receive subscription change notifications on a topic of their choice. 
 
-Dispatchers are also permitted to register for subscription change notifications to facilitate the multicasting of messages. The mechanics and requirements of the dispatcher and uSubscription communication are platform deployment specific. 
+[.specitem,oft-sid="dsn~usubscribe-register-notifications~1",oft-needs="impl,utest"]
+--
+Clients *MAY* use `RegisterForNotifications()` to receive subscription change notification messages on a custom topic of their choice.
+--
 
-== Event Delivery
+[.specitem,oft-sid="dsn~usubscribe-unregister-notifications~1",oft-needs="impl,utest"]
+--
+Clients *MAY* use `UnregisterForNotifications()` to stop receiving subscription change notification messages on a custom topic.
+--
```

Comment:
instead of specifying what a client may do, we should specify what a uSubscription service MUST do, i.e. send notifications to the topic registered by a client ...